File: src/1295_find-numbers-with-even-number-of-digits.py
class Solution:
    def calc_num_digits(self, n):
        num_digits = 0

        while n:
            num_digits += 1
            n //= 10

        return num_digits

    def findNumbers(self, nums: List[int]) -> int:
        count_even = 0

        for num in nums:
            count_even += int(self.calc_num_digits(num) % 2 == 0)

        return count_even

File: Elixir/1295_find-numbers-with-even-number-of-digits.ex
defmodule Solution do
  def num_digits(0), do: 0
  def num_digits(num), do: num_digits(div(num, 10)) + 1

  @spec find_numbers(nums :: [integer]) :: integer
  def find_numbers(nums) do
    Enum.count(nums, &(rem(num_digits(&1), 2) == 0))
  end
end
